### PR TITLE
Create oas

### LINKED ISSUE #10: Create OAS

### PR DESCRIPTION
**Pull request by anacabbar**
fixes #10

### CODE REVIEW

File: spec/MicroWavePerformanceProxy.yaml [hunk 1]
```diff
@@ -439,6 +439,1818 @@ paths:
...
+                            logical-termination-point-id:
+                              type: string
+                              description: >
+                                'Identifier of the ethernet container within the MW SDN domain'
+                        ethernet-container-performance-measurements-list:
+                          type: array
+                          items:
+                            type: object
+                            properties:
+                              granularity-period:
```
Comment (PrathibaJee) [inline spec/MicroWavePerformanceProxy.yaml:827]: A description shall be added to point refer the resource in **$RequestForProvidingHistoricalPmDataCausesReadingHistoricalEthernetContainerPerformanceFromCache.EthernetContainerHistoricalPmFromCache**.response.body#ethernet-container-2-0:ethernet-container-historical-performance/historical-performance-data-list/**granularity-period**.

File: spec/MicroWavePerformanceProxy.yaml [hunk 2]
```diff
@@ -439,6 +439,1818 @@ paths:
...
+                  $ref: '#/components/responses/responseForErroredServiceRequests'
+                '403':
+                  $ref: '#/components/responses/responseForErroredServiceRequests'
+                '404':
+                  $ref: '#/components/responses/responseForErroredServiceRequests'
+                '500':
+                  $ref: '#/components/responses/responseForErroredServiceRequests'
+                default:
+                  $ref: '#/components/responses/responseForErroredServiceRequests'
+        RequestForProvidingHistoricalPmDataCausesReadingAirInterfaceCapabilitiesFromCache:
```
Comment (PrathibaJee) [inline spec/MicroWavePerformanceProxy.yaml:1748]: In the transmission-mode-list(line no 729) ,  we are formulating the attributes, 
```
transmission-mode-name, 
number-of-states, 
modulation-scheme-name-at-lct, 
channel-bandwidth, code-rate, 
symbol-rate-reduction-factor, 
transmit-power-minimum, 
transmit-power-maximum,
receive-threshold,
transmission-mode-rank,
adaptive-modulation-upshift-level,
adaptive-modulation-downshift-level,
xpic-is-available ,
supported-as-fixed-configuration 
```
from this callback **RequestForProvidingHistoricalPmDataCausesReadingAirInterfaceCapabilitiesFromCache.CapabilitiesFromCache**. 

But , in the response of the callback , we have only the attributes 
```
transmission-mode-name, 
channel-bandwidth, 
modulation-scheme, 
modulation-scheme-name-at-lct.
```
The missing attributes shall also be included in the response

File: spec/MicroWavePerformanceProxy.yaml [hunk 3]
```diff
@@ -439,6 +439,1818 @@ paths:
...
+                                description: >
+                                  'If true, the this transmission mode is available for manual configuration when Adaptive Modulation is switched off
+                                  from {$RequestForProvidingHistoricalPmDataCausesReadingAirInterfaceCapabilitiesFromCache.CapabilitiesFromCache.response.body#air-interface-2-0:air-interface-capability/transmission-mode-list/supported-as-fixed-configuration}'
+                  ethernet-container-list:
+                    description: 'List of ethernet containers found in the given devices'
+                    type: array
+                    items:
+                      type: object
+                      properties:
+                        ethernet-container-identifiers:
```
Comment (PrathibaJee) [inline spec/MicroWavePerformanceProxy.yaml:811]: The description of the attributes **ethernet-container-list/interface-name, logical-termination-point-id** shall be updated to point a callback to fomulate the values.

File: spec/MicroWavePerformanceProxy.yaml [hunk 4]
```diff
@@ -439,6 +439,1818 @@ paths:
                   $ref: '#/components/responses/responseForErroredServiceRequests'
                 default:
                   $ref: '#/components/responses/responseForErroredServiceRequests'
+  /v1/provide-historical-pm-data-of-device:
+    post:
+      operationId: provideHistoricalPDataOfDevice
```
Comment (PrathibaJee) [inline spec/MicroWavePerformanceProxy.yaml:444]: operationId shall be modified to provideHistoricalP**M**DataOfDevice

**Review by PrathibaJee — CHANGES_REQUESTED** [review]
Hi Ana , 
The server side stub generated with the provided OAS is working good interms of the request and response schema validation.
Also kindly please find few minor comments. Consider them if they are valid.
Many thanks.

Thanks,
Prathiba J

File: spec/MicroWavePerformanceProxy.yaml (hunk 1 above)
Comment (anacabbar) [reply spec/MicroWavePerformanceProxy.yaml:827]: Description with ressource reference added

File: spec/MicroWavePerformanceProxy.yaml (hunk 3 above)
Comment (anacabbar) [reply spec/MicroWavePerformanceProxy.yaml:811]: Callback reference added in description

File: spec/MicroWavePerformanceProxy.yaml (hunk 2 above)
Comment (anacabbar) [reply spec/MicroWavePerformanceProxy.yaml:1748]: Included the missing attributes in callback response. Also removed some unnecessary ones from both service response and callback response.

**Pull request merged**